pkg/commands/config.go

```diff
@@ -139,6 +159,55 @@ func getBaseImage(platform string, bo *options.BuildOptions) build.GetBase {
 	}
 }
 
+// verifier memoizes digest verification results, so that we avoid redundant
+// digest verification.
+type verifier struct {
+	m       sync.Mutex
+	entries map[name.Digest]*entry
+}
+
+// entry is used to hold the sync.Once used to execute the digest verification,
+// and the resulting error (possibly nil), that results from verification.
+type entry struct {
+	once   sync.Once
+	result error
+}
+
+func (v *verifier) verify(ctx context.Context, digest name.Digest) error {
```

Comment:
I'd like to have a better idea how verification will look in the fullness of time. For now, it sounds like it's just "verify this image has any signature" -- does it also look it up in Rekor? Eventually folks might want to disable checking Rekor, or express other signature requirements.

Even just a sketch of how we envision that looking (in `.ko.yaml`?) would be helpful to me.